docs/test_skip_expiry_design.md

```diff
@@ -0,0 +1,278 @@
+# Design Document: Test Skip Expiry Feature
+
+## Executive Summary
+
+This document proposes a solution to prevent test skips from being forgotten indefinitely by introducing an expiry mechanism. After the expiry date, tests must run and cannot be skipped, ensuring that temporary skips are revisited and either fixed or explicitly renewed.
+
+---
+
+## Table of Contents
+
+1. [Background](#background)
+2. [Problem Statement](#problem-statement)
+3. [Proposed Approaches](#proposed-approaches)
+   - [Approach 1: Inline Expiry Date Field](#approach-1-inline-expiry-date-field)
+   - [Approach 2: Expiry Management via Separate Configuration](#approach-2-expiry-management-via-separate-configuration)
+4. [Comparison and Recommendation](#comparison-and-recommendation)
+5. [Detailed Design (Recommended Approach)](#detailed-design-recommended-approach)
+6. [Implementation Considerations](#implementation-considerations)
+7. [Migration Strategy](#migration-strategy)
+8. [Testing Strategy](#testing-strategy)
+9. [Future Enhancements](#future-enhancements)
+
+---
+
+## Background
+
+The `tests_mark_conditions.yaml` file serves as a centralized location for defining test skips and their conditions. The pytest plugin `conditional_mark` processes this file to automatically skip tests based on platform, topology, release version, and other conditions.
+
+While this consolidation improves maintainability, tests can be skipped and forgotten, leading to:
+- Tests remaining skipped long after issues are resolved
+- Technical debt accumulation
+- Reduced test coverage
+- Loss of visibility into why tests were skipped
+
+---
+
+## Problem Statement
+
+**Goal**: Implement an expiry mechanism to ensure that skipped tests are periodically reviewed and either:
+1. Re-enabled if the underlying issue is resolved
+2. Updated with a new expiry date if the issue persists
+3. Documented with appropriate justification
+
+**Requirements**:
+- Expiry dates should be optional to maintain backward compatibility
+- Clear handling of expired skips (force test to run or fail with explicit message)
+- Timezone-aware date handling for consistency across global teams
+- Minimal disruption to existing workflows
+- Clear reporting of expired skips
+
+---
+
+## Proposed Approaches
+
+### Approach 1: Inline Expiry Date Field
+
+Add an `expiry_date` field directly within each skip/xfail definition in the YAML file.
+
+#### Structure Example
+
+```yaml
+acl/test_acl.py:
+  skip:
+    reason: "Skip acl for isolated-v6 topology"
```

Comment:
Should we have a category why it get skipped? We may allow infinite skip time for specified skip reasons, for example, ASIC not support or TOPO not support? 
For any other skip, it should be skipped by next six months or less, and we may enforce this in PR checker? If someone want to skip, and not fall into the infinite skip categories, if the expiry_date is missing or longer than allowed time, it will be marked as PR checker failure. 
And how we handle if there are two platforms, for same case, one of them can apply infinite skip because the feature is not supported, but the other need to skip with expiry_date, could the skip plugin treat them as two different skip cases and honor both of them but not only one of them?